akka-typed/src/main/scala/akka/typed/patterns/Restarter.scala

```diff
@@ -20,8 +20,9 @@ final case class Restarter[T, Thr <: Throwable: ClassTag](initialBehavior: Behav
   behavior: Behavior[T] = initialBehavior) extends Behavior[T] {
 
   private def restart(ctx: ActorContext[T]): Behavior[T] = {
-    try behavior.management(ctx, PreRestart) catch { case NonFatal(_) ⇒ }
+    try behavior.management(ctx, PreRestart) catch { case NonFatal(_) ⇒ } // TODO shouldn't it stop?
     Behavior.canonicalize(initialBehavior.management(ctx, PreStart), initialBehavior)
+    // TODO since we don't create a new instance of the Behavior we should perhaps have a MutableBehavior that is like Deferred + Stateful
```

Comment:
I tried something for this but couldn't come up with something that was better than `Deferred` apart from possibly more obvious naming.